language/tools/transaction-replay/src/lib.rs

```diff
@@ -0,0 +1,118 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::transaction_debugger_interface::{DebuggerStateView, StorageDebuggerInterface};
+use anyhow::{format_err, Result};
+use libra_types::{
+    account_address::AccountAddress,
+    account_state::AccountState,
+    transaction::{Transaction, TransactionOutput, Version},
+};
+use libra_vm::{LibraVM, VMExecutor};
+use resource_viewer::{AnnotatedAccountStateBlob, MoveValueAnnotator};
+use std::convert::TryFrom;
+
+pub mod transaction_debugger_interface;
+
+pub struct LibraDebugger<S: StorageDebuggerInterface> {
+    debugger: S,
+}
+
+impl<S: StorageDebuggerInterface> LibraDebugger<S> {
+    pub fn new(debugger: S) -> Self {
+        Self { debugger }
+    }
+
+    pub fn execute_transactions_at_version(
+        &self,
+        version: Version,
+        txns: Vec<Transaction>,
+    ) -> Result<Vec<TransactionOutput>> {
+        let state_view = DebuggerStateView::new(&self.debugger, version);
+        LibraVM::execute_block(txns, &state_view)
+            .map_err(|err| format_err!("Unexpected VM Error: {:?}", err))
+    }
+
+    pub fn execute_past_transactions(
+        &self,
+        mut begin: Version,
+        mut limit: u64,
```

Comment:
What value should clients provide if they want to say "replay everything from `begin` to the latest known version" (probably a common use-case)?

Reply:
"client" as in end point users like VASP operators?

Reply:
Yes (or anyone else using this tool).